### PR TITLE
Actually hash view for activesetCache

### PR DESCRIPTION
**Pull request by noamnelke**

### CODE REVIEW

**Review by barakshani — APPROVED** [review]

File: activation/atxdb.go [hunk 1]
```diff
@@ -77,7 +78,7 @@ func (db *ActivationDb) CalcActiveSetFromView(a *types.ActivationTx) (uint32, er
 		return 0, err
 	}
 
-	count, found := activesetCache.Get(common.BytesToHash(viewBytes))
+	count, found := activesetCache.Get(sha256.Sum256(viewBytes))
```
Comment (zalmen) [inline activation/atxdb.go]: sort the view before hashing

**Review by zalmen — CHANGES_REQUESTED** [review]

File: activation/atxdb.go (hunk 1 above)
Comment (noamnelke) [reply activation/atxdb.go]: 01ff5a5

File: activation/activation_test.go [hunk 2]
```diff
@@ -213,9 +215,12 @@ func newBuilder(activationDb ATXDBProvider) *Builder {
 func setActivesetSizeInCache(t *testing.T, activesetSize uint32) {
 	view, err := meshProvider.GetOrphanBlocksBefore(meshProvider.LatestLayer())
 	assert.NoError(t, err)
+	sort.Slice(view, func(i, j int) bool {
```
Comment (antonlerner) [inline activation/activation_test.go:218]: use your function you wrote in calculating atx view hash to get layer hash

File: activation/atxdb.go [hunk 3]
```diff
@@ -124,12 +124,26 @@ func (db *ActivationDb) CalcActiveSetFromView(a *types.ActivationTx) (uint32, er
...
 		return 0, err
 	}
-	activesetCache.Add(common.BytesToHash(viewBytes), counter)
+	activesetCache.Add(viewHash, counter)
 
 	return counter, nil
 
 }
 
+func calcSortedViewHash(atx *types.ActivationTx) ([32]byte, error) {
```
Comment (antonlerner) [inline activation/atxdb.go:183]: add a small UT, to make sue this doesnt brake in the future

Comment (noamnelke) [reply activation/atxdb.go:183]: done

File: activation/activation_test.go (hunk 2 above)
Comment (noamnelke) [reply activation/activation_test.go:218]: As discussed, since this is a test, I'll leave it as is.

**Review by zalmen — APPROVED** [review]

**Pull request merged**